src/clj/puppetlabs/services/jruby/jruby_puppet_service.clj

```diff
@@ -148,10 +149,9 @@
                         "jruby-puppet.max-active-instances.")}))
      (when (jruby-schemas/shutdown-poison-pill? pool-instance#)
        (jruby/return-instance ~jruby-service pool-instance# ~reason)
-       (sling/throw+
-        {:type    ::service-unavailable
-         :message (str "Attempted to borrow a JRuby instance from the pool "
-                       "during a shutdown. Please try again.")}))
+       (mw/throw-service-unavailable!
```

Comment:
I think these helpers are nice but it's unfortunate that they're in a project called `ring-middleware` since it's the helpers themselves don't have anything to do with middleware and without already knowing the connection between the helpers and the middleware, it seems weird out of context.